include/okapi/api/coreProsAPI.hpp

```diff
@@ -94,3 +98,29 @@ class CrossplatformThread {
 
   CROSSPLATFORM_THREAD_T thread;
 };
+
+class CrossplatformMutex {
```

Comment:
Yep, this is what I had in mind. The PROS support for mutex should come around the fall semester.